sre-recipes/recipe.py

```diff
@@ -40,3 +42,72 @@ def verify(self):
         Verifies that the user of the recipe found the root cause
         of the breakage
         """
+
+    @abc.abstractmethod
+    def hint(self):
+        """
+        Provides a hint about the root cause of the issue
+        """
+
+    @staticmethod
+    def _run_command(command):
+        """Runs the given command and returns any output and error"""
+        process = subprocess.Popen(command.split(), stdout=subprocess.PIPE)
+        output, error = process.communicate()
+        return output, error
+
+    @staticmethod
+    def _auth_cluster(cluster="APP"):
+        """ Authenticates for kubectl commands. """
+        logging.info('Authenticating cluster')
```

Comment:
Should we add any error handling here? What if the cluster isn't set up?

Reply:
I set up error handling to 1) make sure there's a project, and 2) make sure there's a cluster to authenticate against. Do you think this is enough?

Reply:
Yeah, looks good!